examples/backend-adapters/server/src/effect.ts

```diff
@@ -1,46 +1,59 @@
-import { HttpServer } from "@effect/platform";
-import { BunHttpServer, BunRuntime } from "@effect/platform-bun";
-import { Effect, Layer } from "effect";
+import "dotenv/config";
 
-import { createRouteHandler } from "uploadthing/server";
+import { createServer } from "node:http";
+import { NodeHttpServer, NodeRuntime } from "@effect/platform-node";
+import * as Http from "@effect/platform/HttpServer";
+import * as Effect from "effect/Effect";
+import * as Layer from "effect/Layer";
+
+import { createRouteHandler } from "uploadthing/effect-http";
 
 import { uploadRouter } from "./router";
 
-const { GET, POST } = createRouteHandler({
+const UploadThingRouter = createRouteHandler({
   router: uploadRouter,
 });
 
-const serverResponseFromWeb = (_: Response) =>
-  HttpServer.response.raw(_.body, {
-    headers: HttpServer.headers.fromInput(Object.entries(_.headers)),
-    status: _.status,
-  });
-
-const router = HttpServer.router.empty.pipe(
-  HttpServer.router.get("/api", HttpServer.response.text("Hello from Effect")),
-
-  HttpServer.router.get(
-    "/api/uploadthing",
-    HttpServer.request.ServerRequest.pipe(
-      Effect.map(BunHttpServer.request.toRequest),
-      Effect.map(GET),
-      Effect.flatMap(serverResponseFromWeb),
-    ),
-  ),
-
-  HttpServer.router.post(
-    "/api/uploadthing",
-    HttpServer.request.ServerRequest.pipe(
-      Effect.map(BunHttpServer.request.toRequest),
-      Effect.andThen(POST),
-      Effect.flatMap(serverResponseFromWeb),
-    ),
-  ),
+/**
+ * Simple CORS middleware that allows everything...
+ * Adjust to your needs.
+ */
+const cors = Http.middleware.make((app) =>
+  Effect.gen(function* () {
+    const req = yield* Http.request.ServerRequest;
+
+    const corsHeaders = {
+      "Access-Control-Allow-Origin": "*",
+      "Access-Control-Allow-Methods": "*",
+      "Access-Control-Allow-Headers": "*",
+    };
+
+    if (req.method === "OPTIONS") {
+      return Http.response.empty({
+        status: 204,
+        headers: Http.headers.fromInput(corsHeaders),
+      });
+    }
+
+    const response = yield* app;
+
+    return response.pipe(Http.response.setHeaders(corsHeaders));
+  }),
+);
+
+const router = Http.router.empty.pipe(
+  Http.router.get("/api", Http.response.text("Hello from Effect")),
+  Http.router.mount("/api/uploadthing", UploadThingRouter),
 );
 
-router.pipe(
-  HttpServer.server.serve(),
-  Layer.provide(BunHttpServer.server.layer({ port: 3000 })),
-  Layer.launch,
-  BunRuntime.runMain,
+const app = router.pipe(
+  cors,
+  Http.server.serve(Http.middleware.logger),
+  Http.server.withLogAddress,
 );
+
+const port = +(process.env.PORT || "3000");
+
+const ServerLive = NodeHttpServer.server.layer(() => createServer(), { port });
```

Comment:
```ts
const Port = Config.integer("PORT").pipe(Config.withDefault(3000))
const ServerLive = Layer.unwrapEffect(
  Effect.map(Port, port => NodeHttpServer.server.layer(() => createServer(), { port })
)

Reply:
Does effect/Config work with process.env, import.meta.env etc? Wondering if we could replace std-env with it?